### PR TITLE
moveit_ros integration tests

### PR DESCRIPTION
**Pull request by rhaschke**
integration tests for #442, #592
imported ROS Industrial's moveit_config for Fanuc robot
`rostest` files launch `move_group` and run tests against it

### CODE REVIEW

**Comment by davetcoleman** [comment]
@gavanderhoorn can you review this since I see your name on some of the files?

**Comment by gavanderhoorn** [comment]
I'd be happy to, but I'm currently OoO, so I can't guarantee that I'll be able to do this soon.

Quick comment: there are still references to the `fanuc_driver` and other packages in the launch files and `run_depends` sections. Might want to take those out, or make things functional again.

**Comment by rhaschke** [comment]
Good point. I cleaned the repo up to remove (hopefully all) unnecessary stuff.
**Please squash before merging!**

File: .travis.yml [hunk 1]
```diff
@@ -1,74 +1,21 @@
-# Generic MoveIt Travis Continuous Integration Configuration File
-# Works with all MoveIt! repositories/branches
-# Author: Dave Coleman, Jonathan Bohren, Robert Haschke
+# This config file for Travis CI utilizes https://github.com/ros-planning/moveit_ci/ package.
```
Comment (davetcoleman) [inline .travis.yml:1]: as @v4hn pointed out, would be better if this was in a separate PR and merged in first

**Comment by davetcoleman** [comment]
Do you have any thoughts on our various testing robots in moveit? Seems like we should consolidate or at least understand what is being used:

PR2:
https://github.com/ros-planning/moveit_resources/tree/master/test/urdf

KukakR210 and Motoman SIA20d:
https://github.com/ros-planning/moveit_kinematics_tests

Even better - I think the urdfs you are using should also be moved to moveit_resources

**Comment by rhaschke** [comment]
> PR2: https://github.com/ros-planning/moveit_resources/tree/master/test/urdf

I didn't know about moveit_resources! Instead of importing a new robot for testing, I suggest to use the existing PR2 model. @davetcoleman: Is there a moveit_config around for it too?

**Comment by v4hn** [comment]
On Fri, Jul 29, 2016 at 01:31:43AM -0700, Robert Haschke wrote:

> I didn't know about moveit_resources! Instead of importing a new robot for testing, I suggest to use the existing PR2 model.

Having a _small_ and _easy_ robot around in the resources and to use it for unit tests sounds like a good idea to me.

I don't see a reason to depend on the whole PR2 for tests.

> @davetcoleman: Is there a moveit_config around for it too?

You mean https://github.com/ros-planning/moveit_pr2/tree/indigo-devel/pr2_moveit_config ?

**Comment by rhaschke** [comment]
> You mean https://github.com/ros-planning/moveit_pr2/tree/indigo-devel/pr2_moveit_config ?

Argh! That's yet another location. I think, there should be a full robot (URDF) and moveit config for testing in one central location. I see the following options:
1. Use `moveit_pr2` for integration testing. However, I agree with @v4hn, that it would be better to have a simpler robot around for testing.
2. Add a moveit config for the PR2 in `moveit_resources`. This seems to be redundant with moveit_pr2 then.
3. Move the `test_robot` config of this PR to `moveit_resources`. In that case, it would make sense to replace the PR2 stuff with the test_robot. Not sure though, were this is already used...

Finally I suggest to rename `moveit_resources` to `moveit_test_resources`.

**Comment by v4hn** [comment]
On Fri, Jul 29, 2016 at 02:04:53AM -0700, Robert Haschke wrote:

> 1. Move the `test_robot` config of this PR to `moveit_resources`.

+1 This sounds like a good idea.

> In that case, it would make sense to replace the PR2 stuff with the test_robot. Not sure though, were this is already used...

This has been around for a long time, so we shouldn't touch the existing files in indigo-devel.
I would propose to add a `test_robot` directory in `moveit_resources/test` for now
and move the pr2 related files to a `pr2` subdirectory, or remove them completely.
Either way, the second step might need some thorough `grep`ing through moveit to check whether anything used the current paths.

> Finally I suggest to rename `moveit_resources` to `moveit_test_resources`.

-1 for indigo. That might make sense for kinetic though.
The current name is a bit confusing, I agree.

**Comment by davetcoleman** [comment]
> I think, there should be a full robot (URDF) and moveit config for testing in one central location. 

moveit_robots is where we've historically kept all moveit config packages, i like the idea of keeping them there. Do you think the ones we share with users will need to be different than the ones we test with? I don't like the idea of having duplicates

> I agree with @v4hn, that it would be better to have a simpler robot around for testing.

Yes I see no problem having a single arm robot alongside PR2 - though I am concerned about having too many robots. See [these ik tests](https://github.com/ros-planning/moveit/issues/4) @jrgnicho 

> Move the test_robot config of this PR to moveit_resources. In that case, it would make sense to replace the PR2 stuff with the test_robot. Not sure though, were this is already used...

The pr2 is being used in [moveit's limited tests already](https://github.com/ros-planning/moveit_planners/blob/kinetic-devel/ompl/ompl_interface/test/test_state_space.cpp#L65) - we should def not remove it but moving it to a better labeled subfolder sounds great

> Finally I suggest to rename moveit_resources to moveit_test_resources.

This is a released package so we can't do that for indigo and I don't think its worth the effort in the future. Just add a better README perhaps

**Comment by rhaschke** [comment]
As a cleanup of moveit_resources will involve adjustments in several other repos, I suggest to postpone this PR until the repos are merged. Closing here in favour of https://github.com/ros-planning/moveit/issues/17.
